### PR TITLE
Added authorization for delete and update pages

### PR DESCRIPTION
**Pull request by amalts0101**
- Added reference fields in posts.
- Changed post controllers.
- Added associations.
- Made changes in index page.

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -1,3 +1,4 @@
+# Application controller
 class ApplicationController < ActionController::Base
-  before_action :authenticate_user!
+  before_action :authenticate_user!, :except => [:index, :show]
```
Comment (eshaiju) [inline app/controllers/application_controller.rb]: use latest hash syntax

Comment (eshaiju) [reply app/controllers/application_controller.rb]: before_action :authenticate_user!, except: [:index, :show]

Comment (amalts0101) [reply app/controllers/application_controller.rb]: ok

**Comment by eshaiju** [comment]
please merge

**Pull request merged**